test/test_indexing.py

```diff
@@ -1041,6 +1043,56 @@ def test_invalid_device(self, device):
         for accumulate in [True, False]:
             self.assertRaises(RuntimeError, lambda: torch.index_put_(b, (idx,), c, accumulate=accumulate))
 
+    def test_take_along_dim(self, device):
+        def _test_against_numpy(t, indices, dim):
+            actual = torch.take_along_dim(t, indices, dim=dim)
+            t_np = t.cpu().numpy()
+            indices_np = indices.cpu().numpy()
+            expected = np.take_along_axis(t_np, indices_np, axis=dim)
+            self.assertEqual(actual, expected, atol=0, rtol=0)
+
+        for shape in [(2, 3, 5), (3, 2), (2, 3, 1, 4)]:
```

Comment:
Are there other degenerate but valid inputs this should check? Like what about tensors with no elements, and noncontiguous tensors (make_tensor() has a "discontiguous" kwarg now to make them easy to test)?